packages/components/src/public/QueryModel/grid/actions/Filter.ts

```diff
@@ -0,0 +1,239 @@
+const TYPE_MAP = Map<string, Filter.IFilterType>(Filter.Types);
+TYPE_MAP.valueSeq().forEach((type: Filter.IFilterType) => {
+    // there are duplicates of the same filter in the type map (e.g. NEQ and NOT_EQUAL)
+    const suffix = getURLSuffix(type);
+    const symbol = type.getDisplaySymbol();
+
+    if (!TYPE_SET.has(suffix)) {
+        TYPE_SET.add(suffix);
+
+        SUFFIX_MAP.set(suffix, type);
+        if (symbol !== null) {
+            if (!SYMBOL_MAP.has(symbol)) {
+                SYMBOL_MAP.set(symbol, Map<string, Filter.IFilterType>().asMutable());
+            }
+
+            SYMBOL_MAP.get(symbol).set(suffix, type);
+        }
+
+        const text = type.getDisplayText();
+        if (text) {
+            TEXT_MAP.set(text.toLowerCase(), type);
+        }
+    }
+});
+SYMBOL_MAP = SYMBOL_MAP.asImmutable();
+SUFFIX_MAP = SUFFIX_MAP.asImmutable();
+TEXT_MAP = TEXT_MAP.asImmutable();
+TYPE_SET = undefined;
+
+/**
+ * Remaps URL suffixes for types where the suffix cannot be distinguished.
+ * @private
+ */
+export function getURLSuffix(type: Filter.IFilterType): string {
+    const suffix = type.getURLSuffix();
+    if (suffix === '') return 'any';
+    return suffix;
+}
+
+/**
+ * Given a symbol/suffix/text and a QueryColumn this will resolve the IFilterType based on the column
+ * type matched against the symbol/suffix/text.
+ * @param token
+ * @param column
+ * @returns {IFilterType}
+ */
+export function resolveFilterType(token: string, column: QueryColumn): Filter.IFilterType {
+    if (SUFFIX_MAP.has(token)) {
+        return SUFFIX_MAP.get(token);
+    }
+
+    if (SYMBOL_MAP.has(token)) {
+        const symbolTypes = SYMBOL_MAP.get(token);
+        const types = Filter.getFilterTypesForType(column.getDisplayFieldJsonType() as JsonType);
+
+        let value: Filter.IFilterType;
+        let match = false;
+
+        for (let i = 0; i < types.length; i++) {
+            const suffix = getURLSuffix(types[i]);
+            if (symbolTypes.has(suffix)) {
+                if (match) {
+                    console.warn(
+                        `Column of type \"${column.get('jsonType')}\" has multiple filters for symbol \"${token}\".`
+                    );
+                    match = false;
+                    value = undefined;
+                    break; // stop the loop, ambiguous
+                } else {
+                    match = true;
+                    value = symbolTypes.get(suffix);
+                }
+            }
+        }
+
+        if (match && value) {
+            return value;
+        }
+    }
+
+    const text = token.toLowerCase();
+    if (TEXT_MAP.has(text)) {
+        return TEXT_MAP.get(text);
+    }
+
+    return undefined;
+}
+
+/**
+ * Given a IFilterType resolves the symbol (string) that should be displayed
+ * @param filterType
+ * @returns {string}
+ */
+function resolveSymbol(filterType: Filter.IFilterType): string {
+    const symbol = filterType.getDisplaySymbol();
```

Comment:
Any chance `filterType` is null here?

Reply:
no idea. this file was just moved. I added a null check though.